src/main/java/org/prebid/server/bidder/thirtythreeacross/ThirtyThreeAcrossBidder.java

```diff
@@ -59,52 +65,88 @@ public TtxBidder(String endpointUrl, JacksonMapper mapper) {
         final Map<String, List<Imp>> impsMap = new HashMap<>();
         final List<HttpRequest<BidRequest>> requests = new ArrayList<>();
 
-        for (Imp imp : request.getImp()) {
+        final BidRequest modifiedRequest = modifyRequest(request, errors);
+        for (Imp imp : modifiedRequest.getImp()) {
             try {
                 validateImp(imp);
-                final ExtImpTtx extImpTtx = parseImpExt(imp);
-                final Imp updatedImp = updateImp(imp, extImpTtx);
+                final ExtImpThirtyThreeAcross extImpThirtyThreeAcross = parseImpExt(imp);
+                final Imp updatedImp = updateImp(imp, extImpThirtyThreeAcross);
                 impsMap.computeIfAbsent(computeKey(updatedImp), k -> new ArrayList<>()).add(updatedImp);
             } catch (PreBidException e) {
                 errors.add(BidderError.badInput(e.getMessage()));
             }
         }
 
         for (List<Imp> imps : impsMap.values()) {
-            requests.add(createRequest(request, imps));
+            requests.add(createRequest(modifiedRequest, imps));
         }
 
         return Result.of(requests, errors);
     }
 
+    private BidRequest modifyRequest(BidRequest request, List<BidderError> errors) {
+        try {
+            return request.toBuilder().ext(modifyRequestExt(request.getExt())).build();
+        } catch (PreBidException e) {
+            errors.add(BidderError.badInput(e.getMessage()));
+            return request;
+        }
+    }
+
+    private ExtRequest modifyRequestExt(ExtRequest ext) throws PreBidException {
+        final Map<String, JsonNode> properties = ObjectUtil.getIfNotNull(ext, ExtRequest::getProperties);
+        final JsonNode extTtxNode = ObjectUtil.getIfNotNull(properties, p -> p.get("ttx"));
+        final ThirtyThreeAcrossExtTtx extTtx;
+        try {
+            extTtx = mapper.mapper().convertValue(extTtxNode, ThirtyThreeAcrossExtTtx.class);
+        } catch (IllegalArgumentException e) {
+            throw new PreBidException(e.getMessage());
+        }
+
+        final ExtRequest modifiedExt = ExtRequest.empty();
+        if (properties != null) {
+            modifiedExt.addProperties(properties);
+        }
+        modifiedExt.addProperty("ttx", mapper.mapper().valueToTree(modifyRequestExtTtx(extTtx)));
+
+        return modifiedExt;
+    }
+
+    private static ThirtyThreeAcrossExtTtx modifyRequestExtTtx(ThirtyThreeAcrossExtTtx extTtx) {
+        final List<ThirtyThreeAcrossExtTtxCaller> callers;
+        if (extTtx == null || CollectionUtils.isEmpty(extTtx.getCaller())) {
+            callers = Collections.singletonList(PREBID_CALLER);
+        } else {
+            callers = new ArrayList<>(extTtx.getCaller());
+            callers.add(PREBID_CALLER);
```

Comment:
ListUtils.union(extTtx.getCaller(), Collections.singletonList(PREBID_CALLER)) and then create ternary